src/Application.vala

```diff
@@ -39,27 +44,74 @@ public class Sideload.Application : Gtk.Application {
             return;
         }
 
-        var ref_file = new FlatpakRefFile (file);
-        main_window = new MainWindow (this, ref_file);
+        hold ();
+        open_file.begin (file);
+    }
+
+    private async void open_file (File file) {
+        FileInfo? file_info = null;
+        try {
+            file_info = yield file.query_info_async (
+                FileAttribute.STANDARD_CONTENT_TYPE,
+                FileQueryInfoFlags.NONE
+            );
+        } catch (Error e) {
+            print ("Unable to query content type of provided file\n");
```

Comment:
Rather than print, it would make more sense to use a `critical ()` or `warning ()` here:

```suggestion
            critical ("Unable to query content type of provided file: %s", e.message);
```